### PR TITLE
Merging fix for IDENTITY-2924

### PR DESCRIPTION
**Pull request by thanujalk**

### CODE REVIEW

File: components/identity/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
     public static final String DEFAULT_IDP_CONFIG = "default";
 
     public static final String WSO2CARBON_CLAIM_DIALECT = "http://wso2.org/claims";
+    public static final String SF_OAUTH2_TOKEN_ENDPOINT = "https://login.salesforce.com/services/oauth2/token";
```
Comment (hasinthaindrajee) [inline components/identity/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java]: Are we ok to have this endpoint hard coded ?

File: components/identity/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnector.java [hunk 2]
```diff
@@ -30,6 +30,7 @@
 import org.json.JSONTokener;
 import org.wso2.carbon.base.ServerConfiguration;
 import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
 import org.wso2.carbon.identity.provisioning.*;
```
Comment (hasinthaindrajee) [inline components/identity/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnector.java]: Remove imports with *

File: components/identity/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnector.java [hunk 3]
```diff
@@ -367,7 +368,10 @@ private String authenticate() throws IdentityProvisioningException {
 
         HttpClient httpclient = new HttpClient();
 
-        PostMethod post = new PostMethod(SalesforceConnectorConstants.OAUTH2_TOKEN_ENDPOINT);
+        String url = configHolder.getValue(SalesforceConnectorConstants.PropertyConfig.OAUTH2_TOKEN_ENDPOINT);
+
+        PostMethod post = new PostMethod(url != null && !url.isEmpty() ?
```
Comment (hasinthaindrajee) [inline components/identity/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnector.java]: You can use StringUtils here

File: components/identity/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/listener/IdentityProviderMgtProvisioningListener.java [hunk 4]
```diff
@@ -106,6 +107,31 @@ public void destroyConnector(String identityProviderName, String tenantDomain)
...
+                            log.debug("Service Provider '" + serviceProvider + "' Provisioning cached entry removed for idp " + identityProviderName);
+                        }
+                    } else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Service Provider '" + serviceProvider + "' Provisioning cached entry not found for idp " + identityProviderName);
+                        }
+                    }
+                }
+            } catch (IdentityApplicationManagementException e) {
+                log.error(e.getMessage(), e);
```
Comment (hasinthaindrajee) [inline components/identity/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/listener/IdentityProviderMgtProvisioningListener.java]: Do not use e.getMesssage()

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 5]
```diff
@@ -653,8 +654,14 @@ private static void buildSalesforceProvisioningConfiguration(IdentityProvider fe
             domainNameProp.setValue(paramMap.get("sf-domain-name"));
         }
 
+        if (paramMap.get("sf-token-endpoint") != null) {
```
Comment (hasinthaindrajee) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Define String constants

File: components/identity/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java (hunk 1 above)
Comment (thanujalk) [inline components/identity/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java]: This is the default value. User can change this using the UI.

**Pull request merged**